86983jwdt/refactor/routing-with-enums

Delicadency opened the pull request.

NacomiTagiera commented on src/components/History/HistoryNav.tsx:
```diff
@@ -19,23 +20,23 @@ export default function HistoryNav() {
           <div className="h-[50px] w-[50px] rounded-full bg-yellowVintage" />
-          <a href="#">5.Pułk Strzelców Konnych</a>
+          <a href={Routes.REGIMENT}>5.Pułk Strzelców Konnych</a>
         </li>
         <li className="flex items-center gap-5 desktop:flex-col">
           <div className="h-[50px] w-[50px] rounded-full bg-greenC" />
-          <a href="#">Akcja Burza</a>
+          <a href={Routes.OPERATION}>Akcja „Burza”</a>
```
`<a>` tags should be replaced with Next.js Link component

Delicadency replied:
You are absolutely right. My focus was on correcting the paths, not on the “bigger picture.” I will correct it tomorrow, today my brain is already out of order. Thanks 😄

Eghizio replied:
@NacomiTagiera it is not within the scope of this task but yeah 👍 
If we can make something better while doing our task and it is not taking too much time then why not.

NacomiTagiera commented on src/routes.ts:
I'd consider using `as const` objects or `const enum` instead of regular enums - they provide the same type safety with smaller bundle size and no runtime overhead. This aligns with modern TS practices while maintaining the same functionality.
https://www.webdong.dev/en/post/enum-const-enum-and-as-const

Delicadency replied:
Ooo... something new for me 😄  I'm just getting into the topic, thanks for the hint @NacomiTagiera. I will check and correct.

Eghizio replied:
Also worth noting: https://www.typescriptlang.org/docs/handbook/enums.html#const-enum-pitfalls

Eghizio replied:
Hmmm. Maybe if we have nested structure then we should use nested object instead of enums?

Example:
`Routes.UNION.HISTORY`
`Routes.UNION.CALENDAR`
`Routes.HISTORY`
`ROUTES.HISTORY.REGIMENT` etc.

Just an idea

Eghizio replied:
Also 😄 
https://devblogs.microsoft.com/typescript/announcing-typescript-5-8/#the---erasablesyntaxonly-option

Eghizio reviewed (approved):
LGTM

Delicadency commented:
@NacomiTagiera @Eghizio Thanks for the feedback and all the tips. They have been very helpful 💛
I changed the enum to object as const and added two additional exports a bit out of the box, but maybe someone will want to use the following.
I did not change the structure of the object to a nested one. It would have added some unnecessary work in my opinion, while the current structure works and seems quite intuitive to me. But the next time I approach the subject I will probably use it.
